### DISCUSSION TITLE
OPCache configuration

### DISCUSSION THREAD

**Discussion by antoniovj1** [serversideup/docker-php#273] [Ideas & Feature Requests]
**Issue Description:**

Over time, I've utilized OPCache to implement substantial performance improvements in various projects (I'll check with my company if I can show some real world benchmarks). Based on the results, I believe it has become an essential component for optimizing our applications. Consequently, I propose that the PHP image should come preconfigured with OPCache, distinguishing between local development, dev, and prod environments. Additionally, providing options to add custom configurations would be highly beneficial, as specific settings (such as preload) or configurations tailored to certain frameworks, like the one recommended by Symfony ([Symfony Performance](https://symfony.com/doc/current/performance.html#use-the-opcache-byte-code-cache)), may be necessary.

To achieve this, a convenient approach might be to allow the creation of files that override the default configuration, something like:
- opecache.ini.local
- opecache.ini.dev
- opecache.ini.prod

What do you think? Can you think of simpler or more practical ways? Do you find any problems with this approach?

**References:**
[PHP Manual - OPCache](https://www.php.net/manual/en/book.opcache.php)

**Comment by jaydrogers** [comment]
I would love to explore this more! Thanks for sharing.

I have a number of open source things to stabilize, then I would like to provide a solid solution for using OPCache. Any other notes or ideas, I'm all ears!
